lib/src/signed_video_h26x_auth.c

```diff
@@ -118,16 +102,23 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
  * Detects if there are any missing SEI messages based on the GOP counter and updates the GOP state.
  */
 static void
-detect_sei_loss_and_sync_gop(signed_video_t *self)
+detect_lost_sei_and_update_gop_state(signed_video_t *self)
 {
   // Get the last GOP counter before updating.
-  uint32_t last_gop_number = self->gop_info->latest_validated_gop;
-  uint32_t exp_gop_number = last_gop_number + 1;
-  uint32_t new_gop_number = self->gop_info->global_gop_counter;
-
+  uint32_t exp_gop_number = self->gop_info->latest_validated_gop + 1;
   // Compare new with last number of GOPs to detect potentially lost SEIs.
+  uint32_t new_gop_number = self->gop_info->global_gop_counter;
   int64_t potentially_missed_gops = (int64_t)new_gop_number - exp_gop_number;
-
+  // If number of |potentially_missed_gops| is negative, we have either lost SEIs together with a
+  // wraparound of |global_gop_counter|, or a reset of Signed Video was done on the camera. The
+  // correct number of lost SEIs is of less importance, since we only want to know If we have lost
+  // any. Therefore, make sure we map the value into the positive side only. It is possible to
+  // signal to the validation side that a reset was done on the camera, but it is still not
+  // possible to validate pending NALUs.
+  if (potentially_missed_gops < 0 && new_gop_number >= exp_gop_number)
```

Comment:
How can this help to detect the difference between a wraparound and a reset? By definition, `potentially_missed_gops < 0` if and only if `exp_gop_number > new_gop_number`, which means that the second condition in this if statement can never be active.
Did you test this somehow?